GoMeet/src/main/webapp/test/spec/spec-vote-meeting-time-dao.js

```diff
@@ -0,0 +1,93 @@
+describe('VoteMeetingTimeDAO - voteMeetingTime', function () {
+  const MEETING_TIME_ID = 'abc123def456';
+  const NON_EXISTENT_ID = 'non-existent-id';
+  const VOTER = 'John Citizen';
+  const QUERY_STRING = 'meetingTimeId=' 
+      + encodeURIComponent(MEETING_TIME_ID) 
+      + '&voters=' 
+      // Note that encodeURIComponenet encodes spaces to %20, 
+      // but URLSearchParam used in VoteMeetingTimeDAO class encodes 
+      // spaces to +. So need to replace that here
+      + encodeURIComponent(VOTER).replace(/%20/g, '+');
+  const INVALID_DATA_QUERY_STRING = 'meetingTimeId=' 
+      + encodeURIComponent(NON_EXISTENT_ID) 
+      + '&voters=' 
+      + encodeURI(VOTER).replace(/%20/g, '+');
+  const RESPONSE_INIT = { // send to doPost not doGet
+    method: 'POST'
+  };
+  const SUCCESS_RESPONSE = 200;
+  const ERROR_RESPONSE = {
+    status: 404,
+    message: 'This will be an error message',
+  };
+
+  beforeEach(function () {
+    // On two consecutive calls, return:
+    // SUCCESS_RESPONSE on first call and ERROR_RESPONSE on second call
+    spyOn(window, 'fetch').and.returnValues(
+        new Response(JSON.stringify(SUCCESS_RESPONSE), RESPONSE_INIT),
+        new Response(JSON.stringify(ERROR_RESPONSE), RESPONSE_INIT)
+    );
+  });
+
+  it('throws an exception if the meetingTimeId param is not provided', async function () {
+    // Note: cannot use expect().toThrow, because the .toThrow executes before call to
+    // async voteMeetingTime is resolved.
+    let meetingTimeId = null;
+    let errorMessage;
+    try {
+      await VoteMeetingTimeDAO.voteMeetingTime(meetingTimeId, VOTER);
+    } catch (error) {
+      errorMessage = error.message;
+    }
+    expect(errorMessage).toEqual(INSUFFICIENT_REQUEST_PARAM);
```

Comment:
Optional - since these lines are repeated in each of the error tests, it could be good to abstract that into a private helper function. Something similar to what I've done at the bottom of this [test file ](https://github.com/googleinterns/step271-2020/pull/34/files#diff-75ddff44f177749d1205dd33040fe302f9350d3ab3a4b488fc3ae1983a45aae0)

Reply:
Yep, that's a good idea! I've changed it 😊